exercises/queen-attack/example.exs

```diff
@@ -1,20 +1,24 @@
 defmodule Queens do
   @type t :: %Queens{white: {integer, integer}, black: {integer, integer}}
-  defstruct white: {0, 3}, black: {7, 3}
+  defstruct [:white, :black]
+  @board_range 0..7
 
   @doc """
   Creates a new set of Queens
   """
-  @spec new() :: Queens.t()
-  @spec new({integer, integer}, {integer, integer}) :: Queens.t()
-  def new(same, same), do: raise(ArgumentError)
+  @spec new(white: {integer, integer} | nil, black: {integer, integer} | nil) :: Queens.t()
+  def new(opts \\ [])
+
+  def new(white: same, black: same),
+    do: raise(ArgumentError)
+
+  def new(opts) do
+    white = opts |> Keyword.get(:white) |> check_range()
+    black = opts |> Keyword.get(:black) |> check_range()
```

Comment:
Then you can have more of the structure (in pseudocode):

1. Collect queen positions from keyword list
2. Make sure they are valid
3. Create blank board
3. Add each queen to the board, when added, they check for the other queen already at that position 
    - this might be an addition private or public function

Reply:
Could this be a good fit for a `with` pipeline or would it be too messy?

Reply:
I find with constructs to be really useful if you have lots of potential branches (errors) to catch.  I'm not sure this is complex enough to really benefit, but it's worth the experiment if you want to try it!